sklearn/utils/testing.py

```diff
@@ -904,16 +901,21 @@ def _get_func_name(func, class_name=None):
     module = inspect.getmodule(func)
     if module:
         parts.append(module.__name__)
-    if class_name is not None:
-        parts.append(class_name)
-    elif hasattr(func, 'im_class'):
+
+    try:
+        # Python 3
+        qualname = func.__qualname__
+        if qualname != func.__name__:
+            parts.append(qualname[:qualname.find('.')])
+    except AttributeError:
+        # Python 2
```

Comment:
We no longer support python2, so I think this part can be removed.

Reply:
Good catch! Will do!